### PR TITLE
Add hreflang and lang attributes when needed

### PR DESCRIPTION
**Pull request by DuaelFr**
Mainly for accessibility reasons, it's good to indicate to the user agent when the language changes.

- when a link points to a page in an other language than the current document, it's recommended to use a `hreflang` attribute. It allows the screen reader user to know that they are going to land in a page in an other language before clicking.
- when some part of the document is in an other language than the document itself, the element should use a `lang` attribute. It allows the screen reader to adapt its speech to the propre language.

### CODE REVIEW

**Comment by DuaelFr** [comment]
Note: I have no idea how Hugo works. I tried to guess but it seems that something is missing. I highly rely on your skills to make this finally work ;)

File: layouts/shortcodes/translations.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <ul class="translation-list">
   {{ range $.Site.Home.AllTranslations }}
-    <li><a href="{{ .Permalink }}">{{ .Language.LanguageName }}</a></li>
+    <li><a href="{{ .Permalink }}" lang="{{ .Language.LanguageCode }}" hreflang="{{ .Language.LanguageCode }}">{{ .Language.LanguageName }}</a></li>
```
Comment (CoralineAda) [inline layouts/shortcodes/translations.html]: Try this:

```
<ul class="translation-list">
  {{ range $.Site.Home.AllTranslations }}
    <li>
      <a href="{{ .Permalink }}" lang="{{ .Language }}" hreflang="{{ .Language }}">
        {{ .Language.LanguageName }}
      </a>
    </li>
{{ end }}
</ul>
```

Comment (DuaelFr) [reply layouts/shortcodes/translations.html]: It works but I found out that the Spanish langcode was not good so I added a new commit to fix it :)

**Comment by DuaelFr** [comment]
Thank you for your review :)

**Comment by CoralineAda** [comment]
Thanks so much!

**Pull request merged**